machine-learning/Dockerfile

```diff
@@ -17,6 +17,41 @@ RUN mkdir /opt/armnn && \
     cd /opt/ann && \
     sh build.sh
 
+# Warning: 57.2Gb of disk space required to pull this image
+# https://github.com/microsoft/onnxruntime/blob/main/dockerfiles/Dockerfile.rocm
+FROM rocm/pytorch:rocm6.1.2_ubuntu22.04_py3.10_pytorch_release-2.1.2 as builder-whl-rocm
+
+WORKDIR /code
+
+ENV PATH /code/cmake-3.27.3-linux-x86_64/bin:${PATH}
+
+# Prepare onnxruntime repository & build onnxruntime
+RUN git clone --single-branch --branch v1.18.1 --recursive "https://github.com/Microsoft/onnxruntime" onnxruntime
+WORKDIR /code/onnxruntime
+# EDIT PR
+# While there's still this PR open, we need to compile on the branch of the PR
+# https://github.com/microsoft/onnxruntime/pull/19567
+RUN git checkout -b v1.18.1
+RUN git remote add pr https://github.com/dmnieto/onnxruntime
+RUN git fetch pr
+RUN git checkout -b fix_rocm_convolution pr/fix_rocm_convolution
+RUN git config --global user.name 'Foo Bar'
+RUN git config --global user.email 'foo@bar.com'
+RUN git rebase v1.18.1
+COPY ./rocm-patch-onnxruntime.patch /tmp/
```

Comment:
You can get the diff from that PR in patch form by adding .patch to the PR link [like so](https://patch-diff.githubusercontent.com/raw/microsoft/onnxruntime/pull/19567.patch). Then you can just apply that and then your patch (or combine them into one file). This is cleaner and safer.